### ISSUE TITLE
Clean Install of IoTEdge 1.0.7 Fails with "IoT Edge is installed in an invalid location. To reinstall, first remove the existing installation using "Uninstall-IoTEdge""

### ISSUE THREAD

**Issue by davekbell** [Azure/iotedge#1236]
## Expected Behavior
Follow the walkthrough guide for installing IoT Edge on Windows here:
https://docs.microsoft.com/en-us/azure/iot-edge/quickstart

Expected to deploy IoT Edge to Win 10 PC (ver 1809, build 17763)

## Current Behavior
Deployment is successful however running Initialize-IoTEdge fails with message:

"IoT Edge is installed in an invalid location. To reinstall, first remove the existing installation using "Uninstall-IoTEdge"

Running "Uninstall-IoTEdge -Force" is successful however any subsequent Deployment or Initialization fails with message:

"IoT Edge is installed in an invalid location. To reinstall, first remove the existing installation using "Uninstall-IoTEdge"

## Steps to Reproduce
1.  Fresh Win10 install, version 1809, build 17763
2. PowerShell as Admin, run . {Invoke-WebRequest -useb aka.ms/iotedge-win} | Invoke-Expression; Deploy-IoTEdge -ContainerOs Windows
3. Deploy completes successfully with message "To Complete Installation, run "Initialize-IoTEdge"
4.  Upon running "Initialize-IoTEdge" it fails with the message:  "IoT Edge is not yet installed.  To install run "Deploy-IoTEdge first"

Running Deploy-IoTEdge puts us back into an endless loop where it succeeds but the subsequent Initialize-IoTEdge command fails and asks to run Deploy-IoTEdge.

### Device (Host) Operating System
Windows Server 2019 and Windows 10 Pro.   Both 1809, 17763.

### Architecture
amd64

### Container Operating System
Windows Containers

### Runtime Versions
1.0.7

#### iotedged
1.0.7

#### Edge Agent
1.0.7

#### Edge Hub
1.0.7

#### Docker
Does not run

## Logs
As per messages in above, on logs generated

**Comment by veyalla** [comment]
This is an issue of the new version of the script unable to cleanup the old installation properly.

After you run uninstall, manually delete any directories named like “iotedge*” under $Env:ProgramFiles. Then follow the installation steps as documented (Deploy... followed by Initialize...).

Once updated to 1.0.7, future updates should work fine without this manual step.

FYI - @arsing

**Comment by arsing** [comment]
@veyalla As discussed offline, if you see files under `$env:ProgramFiles`, then those are from 1.0.7. 1.0.6 did not place anything in Program Files.

It is expected that the uninstaller leaves the directory behind if it contains files that were not put there by the installer. I don't believe there is any reason there *should* be files in that directory that were not placed by the installer, but if that was what it was in your case please tell me what those files are.

Anyway, the error in the OP would not happen from left-over files in ProgramFiles.

---

@davekbell I cannot reproduce the issue.

1. Can you confirm that you started with 1.0.7 from scratch, and didn't have a previous 1.0.6 or earlier install?

1. Do you have any of these directories:

    - `$env:PROGRAMDATA\iotedge`

    - `$env:PROGRAMDATA\iotedge-moby`

    - `$env:PROGRAMDATA\iotedge-moby-data`

    (In case your `$env:PROGRAMDATA` is not `C:\ProgramData`, also check for each corresponding directory under `C:\ProgramData`)

    If so, what's inside them? And does `Uninstall-` delete them?

1. Please run `Deploy-` and `Initialize-` with the `-Verbose` flag and post the output of both.

**Comment by davekbell** [comment]
To follow up on this:  

1) Previously had an install of 1.0.6
2) No directories present after 1.0.6 uninstall.
3) Commands fail as per OP

An additional data point - this was WinServer19 running on a laptop so we also saw the "Battery" warning messaging.

We ran out of time testing and reverted to wiping the machine and starting again.  Successfully installed 1.0.7.  It seems the upgrade path from 1.0.6 -> 1.0.7
